android/java/org/chromium/chrome/browser/crypto_wallet/util/Validations.java

```diff
@@ -32,63 +33,89 @@ public void validate(NetworkInfo selectedNetwork, KeyringService keyringService,
-            // 3. Not one of token's contract addresses
-            // 4. Has valid checksum
-
             String senderAccountAddressLower =
                     senderAccountAddress.toLowerCase(Locale.getDefault());
 
             String receiverAccountAddressLower =
                     receiverAccountAddress.toLowerCase(Locale.getDefault());
-            String receiverAccountAddressUpper =
-                    Utils.maybeHexStrToUpperCase(receiverAccountAddress);
 
             Resources resources = ContextUtils.getApplicationContext().getResources();
 
-            byte[] bytesReceiverAccountAddress = Utils.hexStrToNumberArray(receiverAccountAddress);
-            if (!receiverAccountAddress.isEmpty()
-                    && bytesReceiverAccountAddress.length != VALID_ACCOUNT_ADDRESS_BYTE_LENGTH) {
-                callback.call(resources.getString(R.string.wallet_not_valid_eth_address), true);
-                return;
-            }
-
             if (senderAccountAddressLower.equals(receiverAccountAddressLower)) {
                 callback.call(resources.getString(R.string.wallet_same_address_error), true);
                 return;
             }
 
-            mIsKnowContracts = false;
-            if (mKnownContractAddresses == null) {
-                assert braveWalletService != null;
-                assert blockchainRegistry != null;
-                assert selectedNetwork != null;
-
-                TokenUtils.getAllTokensFiltered(braveWalletService, blockchainRegistry,
-                        selectedNetwork, selectedNetwork.coin, TokenUtils.TokenType.ALL,
-                        (tokens) -> {
-                            fillKnowContracts(tokens);
-                            checkForKnowContracts(receiverAccountAddressLower, callback, resources);
-                        });
-            } else {
-                checkForKnowContracts(receiverAccountAddressLower, callback, resources);
-            }
-            if (mIsKnowContracts) return;
-
-            keyringService.getChecksumEthAddress(receiverAccountAddress, checksum_address -> {
-                if (receiverAccountAddress.equals(checksum_address)
-                        || receiverAccountAddress.isEmpty()) {
-                    callback.call("", false);
-                } else if (receiverAccountAddressLower.equals(receiverAccountAddress)
-                        || receiverAccountAddressUpper.equals(receiverAccountAddress)) {
-                    callback.call(
-                            resources.getString(R.string.address_missing_checksum_warning), false);
+            braveWalletService.getSelectedCoin(coinType -> {
+                if (coinType == CoinType.SOL) {
+                    braveWalletService.isBase58EncodedSolanaPubkey(
+                            senderAccountAddress, success -> {
+                                String responseMsg =
+                                        resources.getString(R.string.invalid_sol_address);
```

Comment:
to avoid pulling string all the time, do `String responseMsg = "";` and pull the string from resources only `if (!success)`. We should minimise resources and CPU usage, in most cases it will be a valid address, but you pull the invalid message string as default in 100% cases.